internal/hook/hook_tracker.go

```diff
@@ -69,14 +69,16 @@ type HookTracker struct {
 	// HookExecutedCnt indicates the number of executed hooks.
 	hookExecutedCnt int
 	// hookErrs records hook execution errors if any.
-	hookErrs []HookErrInfo
+	hookErrs        []HookErrInfo
+	AsyncItemBlocks *sync.WaitGroup
```

Comment:
`BackupWithResolvers` calls this `Stat` by its end, but `BackupWithResolvers` doesn't know details of each itemBlock, it just calls `backupItemBlock` for all itemBlocks.
On the other hand, PVBs in each itemBlock belong to the itemBlock, so looks like this semantic is more rational --- for each `backupItemBlock` call,  start all PVBs of the itemBlock->wait all PVB complete->run post hooks.

However, at present, when `backupItemBlock` exits, the PVBs have not actually completed, `BackupWithResolvers` needs to call hooker's `Stat` to wait them indirectly.

Therefore, I suggest we make the waiting for PVB completion into `backupItemBlock` and call post hooks in the same place. cc @sseago

Reply:
We can do this in 1.16 only, as in 1.15, we don't support concurrent `backupItemBlock`, if we put the wait inside `backupItemBlock`, PVBs among itemBlocks have to be done in sequence, this is not helpful practically.

Reply:
@sseago Just let us know your thinking

Reply:
@Lyndon-Li "Therefore, I suggest we make the waiting for PVB completion into backupItemBlock and call post hooks in the same place."

So right now, we invoke the "wait for PVB completion and then call hooks" with `go kb.handleItemBlockPostHooks` -- is your suggestion to replace the goroutine call with a direct func call? This would effectively mean that the specific `BackupItemBlock` call would not return until 1) PVB processing completes and 2) post-backup hooks are done.

With the current codebase (i.e. concurrent `backupItemBlock` is not yet implemented), that might, in the short term, have a negative performance impact -- but it may make more sense to handle it that way once we have concurrent `backupItemBlock` processing. However, there still may be edge cases where this has a negative performance impact -- in particular, with large clusters with a large number of node agents, each of which enables multiple parallel PVB processing -- the number of concurrent PVBs we can process may exceed the number of parallel ItemBlocks configured, in which case parallel item block processing will become the bottleneck, not allowing full use of the node agent pods to handle many volumes at once.

Reply:
> With the current codebase (i.e. concurrent backupItemBlock is not yet implemented), that might, in the short term, have a negative performance impact

Yes, agree. This is why we suggest to do this only in 1.16 where concurrent itemBlock processing is fully supported.

Reply:
@Lyndon-Li @sseago
I'm going to merge this PR and cherry pick the changes to branch 1.15 to unblock the release 1.15.1.
Let's continue the discussion on issue https://github.com/vmware-tanzu/velero/issues/8516